middleware/CompressionHandler.md

```diff
@@ -0,0 +1,46 @@
+# Compression Handler
+
+A middleware component that requests, detects and decompresses response bodies.
+
+## Requirements
+
+- Add `Accept-encoding: gzip` to request headers.
+- Decompress response bodies that have the header `Content-Encoding: gzip`
+
```

Comment:
New requirement: this is set as a default handler. We should have a way to turn it off in case we have a testing handler.